src/xml/libxml2.cr

```diff
@@ -315,10 +315,10 @@ lib LibXML
 end
 
 LibXML.xmlGcMemSetup(
-  ->GC.free,
-  ->GC.malloc(LibC::SizeT),
-  ->GC.malloc(LibC::SizeT),
-  ->GC.realloc(Void*, LibC::SizeT),
+  ->(ptr) { GC.free(ptr) },
+  ->(size) { GC.malloc(size) },
+  ->(size) { GC.malloc(size) },
+  ->(ptr, size) { GC.realloc(ptr, size) },
```

Comment:
I'm confused with these changes. Why they don't follow the same pattern as the others?

Reply:
I tried those, they produce a compiler error that I didn't understand 🤷

Reply:
```
 > 2 |   LibPCRE.pcre_malloc = ->LibC.malloc(LibC::SizeT)
                               ^
Error: read before assignment to local variable 'arg0'
```